JSONResourceFile.js

```diff
@@ -223,28 +269,44 @@ JSONResourceFile.prototype._calcLocalePath = function(locale) {
  */
 JSONResourceFile.prototype.getResourceFilePath = function(locale, flavor) {
     locale = locale || this.locale;
-    var dir, newPath, localePath;
+    var newPath, localePath;
     var filename = "strings.json";
 
     if (this.project.settings.resourceFileNames && this.project.settings.resourceFileNames["json"]){
         filename = this.project.settings.resourceFileNames["json"];
     }
     if (this.project.options.projectType) {
         var projectType = this.project.options.projectType.split("-");
-        if (projectType[1] === "c" || projectType[1] === "cpp") {
+        //var projectType = this.project.getProjectType().split("-");
+        if (projectType[1] === "c" || projectType[1] === "cpp" || projectType[1] === "dart") {
             filename = this.project.settings.resourceFileNames[projectType[1]];
         }
     }
-
-    localePath = this._calcLocalePath(locale);
-
-    dir = path.join(this.project.target, this.project.getResourceDirs("json")[0] || "resources");
-    newPath = path.join(dir, localePath, filename);
-
+    
+    localePath = this._calcLocalePath(locale, filename);
+    newPath = path.join(this.project.target, localePath);
+    
     this.logger.trace("Getting resource file path for locale " + locale + ": " + newPath);
     return newPath;
 };
 
+var defaultMappings = {
+    "strings.json": {
+        "template": "[dir]/[resourceDir]/[localeDir]/[filename]"
+    }
+}
+
+JSONResourceFile.prototype._isExistMappingData = function(filename) {
+    var jsonMap = this.project.options?.settings?.jsonMap?.mappings[filename];
```

Comment:
The ?. operator only works on later versions of node and we don't transpile this before running.

Reply:
Yes This Optional chaining is not support node v13 and below. and The webOS is already v16. so I think it's ok to used...?

Reply:
You don't know which version of node your customers/open source partners are using. Loctool is designed to run on node v12 and above, but your plugin won't work on the older versions of node.

Reply:
You are right. I updated the codes not to use that operator.